src/fast-path.js

```diff
@@ -245,6 +245,21 @@ FPp.needsParens = function(assumeExpressionContext) {
     return true;
   }
 
+  if (
+    ((parent.type === "ArrowFunctionExpression" && parent.body === node) ||
+     parent.type === "ExpressionStatement") &&
+      startsWithOpenCurlyBrace(node) &&
```

Comment:
can you move this test after all the `node.type === ...`, we first want to make sure we are on the correct node type before doing this expensive logic.